gli/utils.py

```diff
@@ -318,28 +334,36 @@ def download_data(dataset: str, verbose=False):
             print("All data files already exist. Skip downloading.")
         return
 
-    # First, check the local urls.json file.
-    url_file = os.path.join(data_dir, "urls.json")
-    if os.path.exists(url_file):
-        with open(url_file, "r", encoding="utf-8") as fp:
-            url_dict = json.load(fp)
-    else:
-        # Second, try to download and check the global_urls.json file.
-        global_url_file = os.path.join(data_dir, "global_urls.json")
-        _download(GLOBAL_FILE_URL, global_url_file, verbose=verbose)
-        if os.path.exists(global_url_file):
-            with open(global_url_file, "r", encoding="utf-8") as fp:
+    # First, get urls from EC2
+    data_file_url_dict = {}
+    flag = False  # Whether all urls are retrieved from server
```

Comment:
Can we change `flag` to something more informative, such as `url_retrieval_success`?

Reply:
url_retrieval_success = True

if data_file_url_dict[data_file] is None: url_retrieval_success = False